docs/src/real_world.md

```diff
@@ -397,3 +397,29 @@ res = collect_results!(datadir("results"); black_list = bl, subfolders = true)
 select!(res, Not(:path)) # don't show path this time
 ```
 All `missing` entries were adjusted automatically :)
+
+## Defining parameter sets with restrictions
+
+As already demonstrated in the examples above, for functions where the set of input parameters is the same for each simulation run, a basic dictionary can be used to define these parameters.
+However, often some of the parameters or values should only be considered if another parameter is also included in the set or has a specific value.
+The macro [`@onlyif`](@ref) allows to place such restrictions on values and parameters.
+The following dictionary defines values and parameters for a genetic algorithm:
+
+```julia
+ga_parameters = Dict(
+    :population_size => [20,50,100],
+    :selection => ["roulette-selection", "SUS", "tournament-selection", "linear ranking"],
+    :fitness_scaling => @onlyif(:selection in ("SUS", "roulette-selection"), collect(1.0:20.0)),
+    :tournamet_size => @onlyif(:selection == "tournament-selection", collect(2:10)),
+    :chromosome => [:A, @onlyif(begin
+        size_constr = :population_size <= 50
+        select_constr = :selection != "SUS"
```

Comment:
```suggestion
        size_constr = (:population_size <= 50)
        select_constr = (:selection != "SUS")
```

I suggest that this will increase clarity, because typically `varname = :symbol` is interpreted as assignement at first glance.

Reply:
Really? I don't know. I'm pretty used to the fact that `=` has the lowest operator precedence.